frontends/ast/simplify.cc

```diff
@@ -4585,6 +4586,93 @@ void AstNode::expand_genblock(const std::string &prefix)
 			}
 			break;
 
+		case AST_IDENTIFIER:
```

Comment:
It looks like this duplicates two large chunks of non-trivial code. The first comes from earlier in `expand_genblock`, and the second from the existing implicit wire logic in `simplify`. I worry that code duplication will hurt the maintainability of the frontend, especially in an area that can be finicky (as you know). Do you think it would be possible to avoid this duplication, potentially with appropriate helper functions?

Reply:
Those are indeed duplicate of two chunks of code. Making wrappers would make sense. Will take a look whenever I can.